tests/subsys/usb/desc_sections/src/desc_sections.c

```diff
@@ -198,41 +198,49 @@ static void check_endpoint_allocation(struct usb_desc_header *head)
 	}
 }
 
+/* Determine the number of bytes spanned between two linker-defined
+ * symbols that are normally interpreted as pointers.
+ *
+ * Note that these symbols provide the address of packed structures,
+ * so alignment restrictions are not relevant.
+ */
+#define SYMBOL_SPAN(_ep, _sp) (ptrdiff_t)(intptr_t)((uintptr_t)(_ep) - (uintptr_t)(_sp))
+
 static void test_desc_sections(void)
 {
 	struct usb_desc_header *head;
 
 	TC_PRINT("__usb_descriptor_start %p\n", __usb_descriptor_start);
 	TC_PRINT("__usb_descriptor_end %p\n",  __usb_descriptor_end);
-	TC_PRINT("USB Descriptor table size %d\n",
-		 PTR_DIFF(__usb_descriptor_end, __usb_descriptor_start));
+	TC_PRINT("USB Descriptor table span %zd\n",
```

Comment:
shouldn't it be `%td`?
```
       t      A following integer conversion corresponds to a ptrdiff_t argument, or a following n conversion corresponds to a pointer to  a  ptrdiff_t  argu\u2010
              ment.
```
Mind you, you could have casted the result "safely" to an int for this test purposes :)

Reply:
You're correct; I've just never had a need to use `t` before, though I do know `size_t` and `ptrdiff_t` may not be the same.  Simplified by just using `int`, which also avoids problems if a libc is used that doesn't understand the `z` or `t` modifiers.